### PR TITLE
feat: introduce a maintenance page

### PR DESCRIPTION
**Pull request by lukicenturi**
#18

### CODE REVIEW

**Comment by kelsos** [comment]
hmm `package-lock.json` is not needed here can we remove it we have `yarn.lock`

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+BASE_URL=http://localhost:8000
+RECAPTCHA_SITE_KEY=xxx
+RECAPTCHA_SECRET_KEY=xxx
```
Comment (kelsos) [inline .env.example:3]: we don't need secret in the config the backend should take care of it

File: pages/index.vue [hunk 2]
```diff
@@ -23,6 +23,7 @@ privacy,opensource,accounting,asset-management,taxes,tax-reporting`
 
 export default defineComponent({
   name: 'Index',
+  middleware: 'maintenance',
```
Comment (kelsos) [inline pages/index.vue:26]: We don't really need the middleware in index, users should be able to access the landing page even if the whole backend is down.

**Comment by kelsos** [comment]
Hmm @lukicenturi  I tried setting MAINTENANCE=true but for some reason when rendering the process.env is empty and it will redirect out of it, do you have any idea?

**Comment by lukicenturi** [comment]
> hmm `package-lock.json` is not needed here can we remove it since we have `yarn.lock`

Forgot we use yarn :P

File: pages/maintenance.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <img
+      :class="$style.image"
+      alt="rotki maintenance"
+      src="~/assets/img/maintenance.svg"
+    />
+
+    <heading :class="$style.heading">We'll be back soon!</heading>
+
+    <div :class="$style.description">
+      Sorry for the incovenience but we're performing some maintenance at the
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply pages/maintenance.vue:12]: thanks for the trick hahaha

**Comment by lukicenturi** [comment]
Updated @kelsos you can try again. Apparently the env should be defined again at `nuxt.config.js` https://nuxtjs.org/docs/configuration-glossary/configuration-env#the-env-property

**Pull request merged**